html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,53 @@
 <?php
-$limit = $_GET['target'];
+$limit = mb_convert_kana($_GET['target'], 'a', 'UTF-8');
+if(!is_numeric($limit) || $limit <= 0 || preg_match('/^([1-9]\d*|0)\.(\d+)?$/', $limit)) {
+  http_response_code(400);
+  echo ('1以上の整数をターゲット値としてください');
+  exit();
+}
 
 $dsn = 'mysql:dbname=test;host=mysql';
 $dbuser = 'test';
 $dbpassword = 'test';
+
+try {
+  $db = new PDO($dsn,$dbuser,$dbpassword);
+} catch(PDOException $e) {
+  echo 'DB接続エラー: ' . $e->getMessage();
```

Comment:
DB接続エラーのときはレスポンスコード500を返して処理を停止しましょう。

```suggestion
  http_response_code(500);
  $errMsg = 'DB接続エラー: ' . $e->getMessage();
  echo json_encode($errMsg, JSON_UNESCAPED_UNICODE);
  exit();
```

Reply:
ご提案ありがとうございます。実装いたしました！